Edit Profile + Sort Products/Product Type Functionalities

hibatawfeeq opened the pull request:
1. What was done:
- Edit Profile Functionality was partially completed; a page that displays the current user information and, the choice to edit any of the current user information through a user form. Adding the new information to the database is completed by @nmdoran. From the Scrum Board on Trello, you can refer to "Edit Profile" and from the Story Board, you can refer to "Edit Profile" for Release 3. 
- Sort by Products/Product Type was fully completed; once you click on the header of Product or Product Type, it should sort it. The first click sorts it in an ascending order whereas the second click sorts it in a descending order. From the Scrum Board on Trello, you can refer to "Display Products Alphabetically" and from the Story Board, you can refer to "Sort by.." for Release 2. 

2. Areas that need special review:
- Edit Profile - one problem I was facing is how to get the user information for the user that's currently logged in. Please refer to the file index.js, lines 188-197. So far we've been dealing with user ID "1" and "0". You can see that I commented out some code that I think is correct, and the reason why I commented it out is because it actually crashes the localhost that's why I kept it as user ID "1". Would really appreciate some feedback on this.
- Sort by Products/Product Type - N/A.

3. Unit Testing:
- Edit Profile - this code was tested by ensuring that the user form can take special values and won't allow the user to click on "Edit Profile" button unless the input has the correct format. You can see from the code on file editProfile.ejs under the user form, that I added formats for each user input. I tested this by inputting the wrong and correct values in the fields and see what it does. You can test it the same way, by adding for example an invalid email, postal code or name format and also, correct format. The proof that my code is working is that it is visible on localhost and does what it's supposed to do by informing anyone testing it of what is actually going on.
- Sort by Products/Product Type - this code was tested by clicking on the header of the fridge table of "Products" or "Product Type", whatever desire, directly on the localhost. You can test it the same way, by clicking on the headers. The proof that my code is working is that it is visible on localhost and does what it's supposed to do by directly displaying the results.

4. Pictures of Scrum Board and Story Board:
![Screen Shot 2020-11-22 at 2 37 58 PM](https://user-images.githubusercontent.com/71107287/99922163-00668980-2cfd-11eb-8a1f-f2d3af6ccc56.png)
![Screen Shot 2020-11-22 at 7 57 35 PM](https://user-images.githubusercontent.com/71107287/99922165-00ff2000-2cfd-11eb-926d-80d8d14c0fd7.png)

Commit:
Sprint 3 - work

Created edit profile form, created a sort function to sort product type and product.

Commit:
Sprint 3 - work

Commit:
Edit Profile Functionality

Alikpld commented:
Great work! especially on documenting all your changes in the description :)
We can discuss the review area you mentioned during the standup with the rest of the team before we merge this to main!

nmdoran commented on index.js:
```diff
@@ -61,6 +61,22 @@ express()
+  .get('/addProduct', jsonParser, async (req, res) => {
+    try {
+      const client = await pool.connect();
+      const result = await client.query(`SELECT * FROM products where upper(prod_name)=upper('${req.query.searchParam}')AND user_ID IN ('0','1')`);
```
@mdschnei should we update this to get the userID based on google login?

hibatawfeeq replied:
It's completed!

nmdoran commented on index.js:
```diff
@@ -169,6 +185,34 @@ express()
+      //const client = await pool.connect();
+      //const result = await client.query(`SELECT name, email, postal_code, email_freq FROM Users where user_ID = '${userID}'`);
+      const client = await pool.connect();
+      const result = await client.query(`SELECT name, email, postal_code, email_freq FROM Users where user_ID IN ('1')`);
```
See above comment about updated to google userID

hibatawfeeq replied:
Will do thank you!

hibatawfeeq committed:
Merge branch 'main' into editprofile

Commit:
Edit Profile Functionality

Edits that @nmdoran gave me. Thank you! It should now display user information!

Commit:
Merge branch 'editprofile' of https://github.com/nmdoran/MSCI342 into editprofile

Commit:
More changes!

Commit:
Update index.ejs
